src/python/zquantum/core/circuit/_circuit_template.py

```diff
@@ -325,4 +325,167 @@ def load_circuit_ordering(file):
+    """ Function to generate circuit connectivity and circuit layers
+        for 2D quantum processors with sycamore-like connectivity
+    Args:
+        x_dimension (int): number of qubits per row of the array
+        y_dimension (int): number of qubits per column of the array
+    Returns:
+        final_connectivity (zquantum.core.circuit.CircuitConnectivity)
+        circuit_layers (zquantum.core.circuit.CircuitLayers)
+    """
+    connectivity = []
+
+    # two-colors patterns: patterns
+    pattern1 = np.zeros((y_dimension - 1, 2 * x_dimension - 1))
+    pattern2 = np.zeros((y_dimension - 1, 2 * x_dimension - 1))
+
+    for m in range(0, y_dimension - 1):
+        pattern2[m, :] = m % 2
+
+    py_index = 0
+    for m, y_index in enumerate(range(1, y_dimension, 2)):
+        row_up = []
+        row_down = []
+        px_index = 0
+        for x_index in range(0, x_dimension):
+            node = y_index * x_dimension + x_index
+            if x_index == x_dimension - 1:
+                row_up.append((node, node - x_dimension))
+                pattern1[py_index, px_index] = (x_index + m) % 2 
+                if y_dimension - 1 > y_index:
+                    row_down.append((node, node + x_dimension))
+                    pattern1[py_index + 1, px_index] = (x_index + m + 1) % 2
+                px_index += 1
+            else:
+                row_up.extend([(node, node - x_dimension), (node, node - x_dimension + 1)])
+                pattern1[py_index, px_index] = (x_index + m) % 2 
+                pattern1[py_index, px_index + 1] = (x_index + m) % 2 
+                if y_dimension - 1 > y_index:
+                    row_down.extend([(node, node + x_dimension), (node, node + x_dimension + 1)])
+                    pattern1[py_index + 1, px_index] = (x_index + m + 1) % 2 
+                    pattern1[py_index + 1, px_index + 1] = (x_index + m + 1) % 2 
+                px_index += 2
+        py_index += 2
+
+        connectivity.append(row_up)
+        connectivity.append(row_down)
+
+    # three-color patterns: masks
+    masks = []
+    for n in range(0, 2):
+        mask1 = pattern1.copy()
+        mask2 = pattern2.copy()
+        for m in range(mask1.shape[0]):
+            mask1[m, (m+n)%2:mask1.shape[1]:2] = -1
+            mask2[m, (m+n)%2:mask1.shape[1]:2] = -1
+        masks.extend([mask1, mask2])
+
+    # generate layers
+    layers = []
+    for n, mask in enumerate(masks):
+        layer1 = []
+        layer2 = []
+        for x in range(mask.shape[1]):
+            for y in range(mask.shape[0]):
+                if mask[y, x] == 0:
+                    layer1.append(connectivity[y][x])
+                elif mask[y, x] == 1:
+                    layer2.append(connectivity[y][x])
+        layers.extend([layer1, layer2])
+    
+    final_connectivity = []
+    for group in connectivity:
+        final_connectivity.extend(group)
+
+    return CircuitConnectivity(final_connectivity), CircuitLayers(layers)
```

Comment:
I will defer to your judgment here, but this function is a bit long. Should we consider using some helper functions instead?

Reply:
It is, but I think the structure and logic is pretty clear and splitting it up for the sake of splitting it might not be beneficial.